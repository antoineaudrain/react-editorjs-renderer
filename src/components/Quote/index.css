.er-quote__blockquote {
    margin: 0;
    padding-left: 40px;
}

.er-quote__inner {
    border-left: 3px solid #C4C4C4;
    padding: 10px 12px 10px 40px;
}

.er-quote__content {
    line-height: 1.6em;
    margin: 0 0 10px 0;
    font-size: 15px;
    color: #313649;
    letter-spacing: .005em;
}

.er-quote__caption {
    color: #6E798C;
    margin: 0;
    font-size: 15px;
    line-height: 1.6em;
    letter-spacing: .005em;
}
