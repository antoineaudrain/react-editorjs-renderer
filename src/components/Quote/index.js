import React from 'react'
import PropTypes from 'prop-types'
import ReactHtmlParser from 'react-html-parser'
import propsConstructor from '../../tools/propsConstructor'
import './index.css'

const Quote = ({ data, style = {} }) => {
  const getProps = propsConstructor(style)
  const blockquoteStyle = getProps('quote', 'blockquote')
  const quoteInner = getProps('quote', 'inner')
  const contentStyle = getProps('quote', 'content')
  const captionStyle = getProps('quote', 'caption')

  const getAlignment = () => (data.alignment === 'center' ? 'center' : 'left')

  const content = ReactHtmlParser(data.text)
  const caption = ReactHtmlParser(data.caption)

  return (
    <blockquote {...blockquoteStyle}>
      <div {...quoteInner}>
        <p {...contentStyle} style={{ textAlign: getAlignment() }}>{content}</p>
        {caption && (
          <p {...captionStyle} style={{ textAlign: getAlignment() }}>
            <small>
              <i>{caption}</i>
            </small>
          </p>
        )}
      </div>
    </blockquote>
  )
}

Quote.propTypes = {
  data: PropTypes.shape({
    text: PropTypes.string.isRequired,
    caption: PropTypes.string,
    alignment: PropTypes.oneOf(['left', 'center'])
  }).isRequired,
  style: PropTypes.shape({
    blockquote: PropTypes.object,
    inner: PropTypes.object,
    content: PropTypes.object,
  })
}

export default Quote
