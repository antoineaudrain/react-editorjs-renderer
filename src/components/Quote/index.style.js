const style = {
  blockquote: {
    margin: '0',
    padding: '.4em 0 .4em 40px'
  },
  inner: {
    borderLeft: '3px solid #C4C4C4',
    padding: '10px 12px 10px 40px'
  },
  content: {
    lineHeight: '1.6em',
    margin: '0 0 10px 0',
    fontSize: '15px',
    color: '#313649',
    letterSpacing: '.005em'
  },
  caption: {
    color: '#6E798C',
    margin: '0',
    fontSize: '15px',
    lineHeight: '1.6em',
    letterSpacing: '.005em'
  }
}

export default style