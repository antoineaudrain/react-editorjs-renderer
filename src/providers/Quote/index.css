.er-quote__blockquote {
    margin: 0;
    padding-left: 40px;
}

.er-quote__inner {
    border-left: 3px solid #C4C4C4;
    padding: 10px 12px 10px 40px;
}

.er-quote__content {
    line-height: 1.6em;
    margin: 0 0 10px 0;
}

.er-quote__caption {
    color: #6E798C;
    margin: 0;
}
