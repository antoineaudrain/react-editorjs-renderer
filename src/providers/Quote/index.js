import React from 'react'
import PropTypes from 'prop-types'
import ReactHtmlParser from 'react-html-parser'
import providerPropsConstructor from '../../utils/providerPropsConstructor'
import defaultStyle from './index.style'

const Quote = ({ data, style = {}, config = {} }) => {
  const getProps = providerPropsConstructor(defaultStyle, style, config)
  const blockquoteStyle = getProps('blockquote')
  const quoteBoxStyle = getProps('quote-box')

  const getAlignment = () => (data.alignment === 'center' ? 'center' : 'left')

  const contentStyle = {
    ...getProps('content'),
    textAlign: getAlignment()
  }
  const captionStyle = {
    ...getProps('caption'),
    textAlign: getAlignment()
  }

  const content = ReactHtmlParser(data.text)
  const caption = ReactHtmlParser(data.caption)

  return (
    <blockquote {...blockquoteStyle}>
      <div {...quoteBoxStyle}>
        <p {...contentStyle}>{content}</p>
        {caption && (
          <p {...captionStyle}>
            <small>
              <i>{caption}</i>
            </small>
          </p>
        )}
      </div>
    </blockquote>
  )
}

Quote.propTypes = {
  data: PropTypes.shape({
    text: PropTypes.string.isRequired,
    caption: PropTypes.string,
    alignment: PropTypes.oneOf(['left', 'center'])
  }).isRequired,
  style: PropTypes.object,
  config: PropTypes.object
}

export default Quote
