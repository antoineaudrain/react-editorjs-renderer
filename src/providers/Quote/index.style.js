const style = {
  blockquote: {
    margin: 0,
    paddingLeft: 40
  },
  'quote-box': {
    borderLeft: '3px solid #C4C4C4',
    padding: '10px 12px 10px 40px'
  },
  content: {
    lineHeight: '1.6em',
    margin: '0 0 20px 0'
  },
  caption: {
    color: '#6E798C'
  }
}

export default style
