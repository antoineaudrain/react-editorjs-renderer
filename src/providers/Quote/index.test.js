import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Quote from './index'

configure({ adapter: new Adapter() })

describe('Paragraph provider:', () => {
  const captionlessQuote = shallow(
    <Quote data={{ text: 'test', alignment: 'left' }} />
  )
  const quote = shallow(
    <Quote data={{ text: 'test', caption: 'test', alignment: 'center' }} />
  )

  it('should match snapshot', () => expect(captionlessQuote).toMatchSnapshot())
  it('should match snapshot', () => expect(quote).toMatchSnapshot())
})
