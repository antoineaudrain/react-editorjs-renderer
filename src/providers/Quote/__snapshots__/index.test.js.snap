// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph provider: should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Paragraph provider: should match snapshot 2`] = `ShallowWrapper {}`;
